crud/delete.lua

```diff
@@ -51,10 +54,58 @@ local function call_delete_on_router(space_name, key, opts)
         return nil, DeleteError:new("Space %q doesn't exist", space_name), true
     end
 
+    local primary_index = space.index[0]
+    local space_format = space:format()
+
     if box.tuple.is(key) then
         key = key:totable()
     end
 
+    local ddl_sharding_key = sharding.get_ddl_sharding_key(space_name)
+    if ddl_sharding_key ~= nil then
+        local sharding_key_fieldno_map, err = utils.get_keys_fieldno_map(space_format, ddl_sharding_key)
+        if next(sharding_key_fieldno_map) == nil then
+            return nil, err
+        end
+
+        -- Make sure fields used in sharding key are present in primary index
+        -- TODO: how to invalidate sharding_key_missed_in_index?
+        -- luacheck: globals sharding_key_missed_in_index
+        if sharding_key_missed_in_index == nil then
+            local primary_index_fieldno = utils.get_index_fieldno_map(primary_index)
+            for fieldno in pairs(sharding_key_fieldno_map) do
+                if primary_index_fieldno[fieldno] == false then
+                    sharding_key_missed_in_index = true
+                    break
+                end
+            end
+        end
+        if sharding_key_missed_in_index == true and opts.bucket_id == nil then
+            return nil, DeleteError:new("Sharding key is missed in primary index, specify bucket_id"), true
```

Comment:
AFAIK `true` as the third argument means `needs_reload`. I'm not sure that reload is needed here. We use "external" schema provider and it's completely clear if something goes wrong and reload shouldn't change anything I assume.

Reply:
removed `true`